scripts/notes.ts

```diff
@@ -0,0 +1,239 @@
+  data.forEach((pr: PRDetails) => {
+    if (!pr.labels?.length) return;
+
+    const noteData: any[] = [];
+    const prImpacts: string[] = [];
+    const prType: string[] = [];
+
+    pr.labels.forEach((label) => {
+      const t = label.match(/Type: (\w+)/i);
+      const i = label.match(/Impacts: (\w+)/i);
+
+      if (t) prType.push(t[1].toLowerCase());
+      if (i) prImpacts.push(i[1].toLowerCase());
+    });
+
+    if (prImpacts.length < 1 || prType.length !== 1) {
+      console.error(
+        '\nPRs are required to have at least one Impacts: label and at exactly one Type: label.'
+      );
+      console.log(pr);
+      console.log();
+      process.exit(1);
+    }
+
+    prImpacts.forEach((i) => {
+      noteData.push([i, prType[0], pr.title, pr.ghpr]);
+    });
+
+    notes.push(...noteData);
+  });
+
+  return notes.sort();
+};
+
+const makeNotesMD = (notes: any[]): string => {
+  let md = '';
+  let lastSystem = '';
+  let lastType = '';
+
+  const upCase = (s: string): string => {
+    return s.charAt(0).toUpperCase() + s.slice(1);
+  };
+
+  notes.forEach((note) => {
+    const [sys, typ, title, num] = note;
+    const typIcon = icon[typ];
+    const sameSystem = lastSystem === sys;
+    const sameType = lastType === typ;
+    lastSystem = sys;
+    lastType = typ;
+
+    if (sameType && sameSystem) {
+      md += `- ${title} (#${num})\n`;
+    }
+    if (!sameType && sameSystem) {
+      md += `### ${typIcon} ${upCase(typ)}\n`;
+      md += `- ${title} (#${num})\n`;
+    }
+    if (!sameSystem || (!sameSystem && !sameType)) {
+      // the first item from this group, let's print a title and the first type
+      const url = theme[sys].urlNpm ? theme[sys].urlNpm : 'https://design.cms.gov';
+      md += `## [${theme[sys].longName}](${url}) [${versions[sys]}]\n`;
+      md += `### ${typIcon} ${upCase(typ)}\n`;
+      md += `- ${title} (#${num})\n`;
+    }
+  });
+  return md + '\n';
+};
+
+/**
+ * Display jira links with associated title to copy/paste
+ */
+const displayJiraTickets = (data: PRDetails[]) => {
+  console.log(`\n-- ${c.green('JIRA Tickets')} --`);
+  const notes = data
+    .filter((pr) => {
+      return !pr.ticket?.toLowerCase().includes('ticket');
+    })
+    .map((pr) => {
+      return { ticket: `https://jira.cms.gov/browse/${pr.ticket}`, title: pr.title };
+    });
+  const unticketed = data.filter((pr) => {
+    return pr.ticket?.toLowerCase().includes('ticket');
+  });
+  notes.forEach((note) => console.log(`${note.ticket} - ${note.title}`));
+  console.log(`\n-- ${c.yellow('Unticketed')} --`);
+  unticketed.forEach((note) => console.log(`${note.author} - ${note.ticket} - ${note.title}`));
+};
+
+/**
+ * Writes notes to fs and uses gh-cli to create notes for latest @cmsgov tag
+ */
+const publishNotes = (notes: string) => {
+  const fn = `${versions.core}-release-notes.md`;
+
+  try {
+    writeFileSync(fn, notes, { encoding: 'utf8' });
+  } catch (err) {
+    console.error(err);
+    process.exit(1);
+  }
```

Comment:
What's the difference between try-catching and exiting vs just letting it throw its error and exit node itself?

Reply:
Oh nothing, this is just a little messy, writeFileSync errors should be fine..